kfs-core/src/main/java/edu/arizona/kfs/sys/batch/service/impl/BuildingImportServiceImpl.java

```diff
@@ -220,75 +218,58 @@ private boolean isValid(ArchibusBuildings archBuilding, PrintWriter outReportErr
-			valid = false; 
-			reportMsg += "BuildingAddressCityName(" + archBuilding.getBuildingAddressCityName() + ") is Not Valid, ";
-		}
-		if (isAttributeLenToLong(maxBuildingAddressCityNameLen, archBuilding.getBuildingAddressCityName().length())) {
-			valid = false;
-			reportMsg += "BuildingAddressCityName(" + archBuilding.getBuildingAddressCityName() + ") is longer than " + maxBuildingAddressCityNameLen + ", ";
-		}
-		if (isAttributeNotValid(isBuildingAddressStateCodeRequired, archBuilding.getBuildingAddressStateCode())) {
-			valid = false; 
-			reportMsg += "BuildingAddressStateCode(" + archBuilding.getBuildingAddressStateCode() + ") is Not Valid, ";
-		}
-		if (isAttributeLenToLong(maxBuildingAddressStateCodeLen, archBuilding.getBuildingAddressStateCode().length())) {
-			valid = false;
-			reportMsg += "BuildingAddressStateCode(" + archBuilding.getBuildingAddressStateCode() + ") is longer than " + maxBuildingAddressStateCodeLen + ", ";
-		}
-		if (isAttributeNotValid(isBuildingAddressZipCodeRequired, archBuilding.getBuildingAddressZipCode())) {
-			valid = false; 
-			reportMsg += "BuildingAddressZipCode(" + archBuilding.getBuildingAddressZipCode() + ") is Not Valid, ";;
+		Integer maxBuildingAddressCountryCodeLen = ddService.getAttributeMaxLength(Building.class, "buildingAddressCountryCode");
+
+		if (!isAttributeValid(isBuildingCodeRequired, archBuilding.getBuildingCode(), archBuilding.getBuildingCode().length(), maxBuildingCodeLen, reportMsg, outReportErrorWriter, "BuildingCode(") || 
+			!isAttributeValid(isBuildingNameRequired, archBuilding.getBuildingName(), archBuilding.getBuildingName().length(), maxBuildingNameLen, reportMsg, outReportErrorWriter, "BuildingName(") || 
+			!isAttributeValid(isBuildingStreetAddressRequired, archBuilding.getBuildingStreetAddress(), archBuilding.getBuildingStreetAddress().length(), maxBuildingStreetAddressLen, reportMsg, outReportErrorWriter, "BuildingStreetAddress(") || 
+			!isAttributeValid(isBuildingAddressCityNameRequired, archBuilding.getBuildingAddressCityName(), archBuilding.getBuildingAddressCityName().length(), maxBuildingAddressCityNameLen, reportMsg, outReportErrorWriter, "BuildingAddressCityName(") || 
+			!isAttributeValid(isBuildingAddressStateCodeRequired, archBuilding.getBuildingAddressStateCode(), archBuilding.getBuildingAddressStateCode().length(), maxBuildingAddressStateCodeLen, reportMsg, outReportErrorWriter, "BuildingAddressStateCode(") || 
+			!isAttributeValid(isBuildingAddressZipCodeRequired, archBuilding.getBuildingAddressZipCode(), archBuilding.getBuildingAddressZipCode().length(), maxBuildingAddressZipCodeLen, reportMsg, outReportErrorWriter, "BuildingAddressZipCode(") || 
+			!isAttributeValid(isBuildingAddressCountryCodeRequired, archBuilding.getBuildingAddressCountryCode(), archBuilding.getBuildingAddressCountryCode().length(), maxBuildingAddressCountryCodeLen, reportMsg, outReportErrorWriter, "BuildingAddressCountryCode(")) {
+
+			return false;
```

Comment:
This if statement seems very long.  Is there a reason that you changed it from how it was done in KFS3?  (Errors, advantages)